File: src/engine/logmessages/sessionlogmessage.cpp
#include "sessionlogmessage.h"

#include "diskutil.h"


SessionLogMessage::SessionLogMessage(const QString &source, const QString &message) :
    LogMessage(),
    m_source(source),
    m_message(message) {
}

SessionLogMessage::~SessionLogMessage() {
}

void SessionLogMessage::log() {

    DiskUtil::appendToLogFile("sessions", m_source.leftJustified(16) + m_message);
}

File: src/log_messages/session_log_message.rs
use super::LogMessage;

pub struct SessionLogMessage {
    source: String,
    message: String,
}

impl SessionLogMessage {
    pub fn new(source: String, message: String) -> Self {
        Self { source, message }
    }
}

impl LogMessage for SessionLogMessage {
    fn get_log(&self) -> String {
        "sessions".to_owned()
    }

    fn get_message(&self) -> String {
        format!("{:16} {}", self.source, self.message)
    }
}
